diff --git a/node_modules/@docusaurus/theme-search-algolia/lib/theme/SearchBar/index.js b/node_modules/@docusaurus/theme-search-algolia/lib/theme/SearchBar/index.js
index 878dc9b..815f47d 100644
--- a/node_modules/@docusaurus/theme-search-algolia/lib/theme/SearchBar/index.js
+++ b/node_modules/@docusaurus/theme-search-algolia/lib/theme/SearchBar/index.js
@@ -6,6 +6,7 @@
  */
 import React, {useState, useRef, useCallback, useMemo} from 'react';
 import {createPortal} from 'react-dom';
+import debounce from 'lodash/debounce';
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import {useHistory} from '@docusaurus/router';
 import {useBaseUrlUtils} from '@docusaurus/useBaseUrl';
@@ -131,7 +132,10 @@ function DocSearch({contextualSearch, externalUrlRegex, ...props}) {
         'docusaurus',
         siteMetadata.docusaurusVersion,
       );
-      return searchClient;
+      return {
+        ...searchClient,
+        search: debounce(searchClient.search, 500)
+      };
     },
     [siteMetadata.docusaurusVersion],
   );
